app/controllers/signin_required_authorizations_controller.rb

```diff
@@ -0,0 +1,40 @@
+class SigninRequiredAuthorizationsController < Doorkeeper::AuthorizationsController
+  def new
+    if pre_auth.authorizable?
+      if skip_authorization? || matching_token?
+        if user_has_signin_permission_to_application?
+          auth = authorization.authorize
```

Comment:
Lost comment due to rebase:

> @Davidslv: 
>> you might want to call super here instead of repeating code.

> @h-lame: 
>> If I do that then it'll run these lines of code again:
>>
>> ```ruby
>> if pre_auth.authorizable?
>>   if skip_authorization? || matching_token?
>> ```
>>
>> And I don't know if those are "cheap" or have no side-effects, so while it's annoying to duplicate the code, I'd rather keep this as-is.